pkger/parser.go

```diff
@@ -1152,14 +1152,14 @@ func (p *Pkg) eachResource(resourceKind Kind, minNameLen int, fn func(o Object)
 			continue
 		}
 
-		if len(k.Name()) < minNameLen {
+		if !isDomainName(k.Name()) {
 			pErr.append(resourceErr{
 				Kind: k.Kind.String(),
 				Idx:  intPtr(i),
 				ValidationErrs: []validationErr{
 					objectValidationErr(fieldMetadata, validationErr{
 						Field: fieldName,
-						Msg:   fmt.Sprintf("must be a string of at least %d chars in length", minNameLen),
+						Msg:   "must be a between 3-64 characters long and may only consist of alphanumeric, dots, underderscores, and hyphen characters.",
```

Comment:
```suggestion
						Msg:   "must be between 3-64 characters long and may only consist of alphanumeric, dots, underderscores, and hyphen characters.",
```

might be nice to link to official docs on dns name compliance. might not be though if the context is wrong 🤷

Reply:
I'm hesitant to link docs from k8s here. I'll link them when this gets added to our internal docs though.